src/operations/utils.js

```diff
@@ -0,0 +1,111 @@
+import bitcoinjs from 'bitcoinjs-lib'
+import RIPEMD160 from 'ripemd160'
+import bigi from 'bigi'
+
+export const DUST_MINIMUM = 5500
+
+export function hash160(buff: Buffer) {
+  const sha256 = bitcoinjs.crypto.sha256(buff)
+  return (new RIPEMD160()).update(sha256).digest()
+}
+
+export function hash128(buff: Buffer) {
+  return Buffer.from(bitcoinjs.crypto.sha256(buff).slice(0, 16))
+}
+
+
+// COPIED FROM coinselect, because 1 byte matters sometimes.
+// baseline estimates, used to improve performance
+const TX_EMPTY_SIZE = 4 + 1 + 1 + 4
+const TX_INPUT_BASE = 32 + 4 + 1 + 4
+const TX_INPUT_PUBKEYHASH = 107
+const TX_OUTPUT_BASE = 8 + 1
+const TX_OUTPUT_PUBKEYHASH = 25
+
+function inputBytes(input) {
+  return TX_INPUT_BASE + (input.script && input.script.length > 0 ?
+                          input.script.length : TX_INPUT_PUBKEYHASH)
+}
+
+function outputBytes(output) {
+  return TX_OUTPUT_BASE + (output.script ? output.script.length : TX_OUTPUT_PUBKEYHASH)
+}
+
+function transactionBytes(inputs, outputs) {
+  return TX_EMPTY_SIZE +
+    inputs.reduce((a, x) => (a + inputBytes(x)), 0) +
+    outputs.reduce((a, x) => (a + outputBytes(x)), 0)
+}
+
+//
+
+export function estimateTXBytes(txIn : bitcoinjs.Transaction | bitcoinjs.TransactionBuilder,
+                                additionalInputs : number,
+                                additionalOutputs : number) {
+  let innerTx = txIn
+  if (txIn instanceof bitcoinjs.TransactionBuilder) {
+    innerTx = txIn.tx
+  }
+  const dummyInputs = new Array(additionalInputs)
+  dummyInputs.fill(1)
+  const dummyOutputs = new Array(additionalOutputs)
+  dummyOutputs.fill(1)
+
+  const inputs = [].concat(innerTx.ins, dummyInputs)
+  const outputs = [].concat(innerTx.outs, dummyOutputs)
+
+  return transactionBytes(inputs, outputs)
+}
+
+export function sumOutputValues(txIn : bitcoinjs.Transaction | bitcoinjs.TransactionBuilder) {
+  let innerTx = txIn
+  if (txIn instanceof bitcoinjs.TransactionBuilder) {
+    innerTx = txIn.tx
+  }
+
+  return innerTx.outs.reduce((agg, x) => agg + x.value, 0)
+}
+
+export function decodeB40(input: string) {
+  const characters = '0123456789abcdefghijklmnopqrstuvwxyz-_.+'
+  const base = bigi.valueOf(40)
+  return input.split('').reverse()
```

Comment:
This return statement is pretty large - making it more verbose would probably improve readability.

Reply:
This is more a style thing.

The type checking stuff is important though because some of the type annotations in the new code don't match the behavior of the code.

Reply:
I appreciate this -- will tease this apart a little bit and add some comments here. I let my `.reduce()` magic get a little away from me on this one :smile: